### PR TITLE
PanelEditor: adds annotations and alertState to Redux

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
This PR tries to address: `When entering edit mode for a panel with annotations or alert the annotations or alert state will not load in editor until you refresh (trigger query). A bit tricky to fix.` from #21111.
**Which issue(s) this PR fixes**:
Relates #21111

**Special notes for your reviewer**:
If this is feels like an ok solution, then I'll add tests and such.

### CODE REVIEW

File: public/app/features/dashboard/dashgrid/PanelChromeAngular.tsx [hunk 1]
```diff
@@ -90,20 +92,8 @@ export class PanelChromeAngularUnconnected extends PureComponent<Props, State> {
...
-      // will not automatically re render
-      this.setState({ alertState: undefined });
-    } else if (payload && alertState) {
-      this.setState({ alertState: undefined });
-    } else {
-      // only needed for detecting title updates right now fix before 7.0
-      this.forceUpdate();
-    }
+    // only needed for detecting title updates right now fix before 7.0
+    this.forceUpdate();
```
Comment (hugohaggmark) [inline public/app/features/dashboard/dashgrid/PanelChromeAngular.tsx:96]: Not at all sure about this one, don't know the consequences of doing this without conditions.

**Comment by dprokop** [comment]
Think this is a good move to push annotations to Redux. @torkelo any other thoughts about this?

**Comment by torkelo** [comment]
Think it looks promising, but only looked at it briefly. But want to think about this more as it would be the first time we store panel data in redux and annotations (and alert state) is a big missing component in our new panel & data infrastructure. 

Need to figure out how a panel plugin would subscribe to annotations and etc and how it fits in with panel events / subscriptions:
https://github.com/grafana/grafana/pull/21010 

But with so many 7.0 things going on I would like to pause this

**Comment by torkelo** [comment]
ah, misunderstood this actually fixes a minor bug not only solves it cleaner (I read it first as being a kind of state => redux issue to clean up logic)

**Comment by hugohaggmark** [comment]
Pausing this

**Comment by hugohaggmark** [comment]
Closing this for now until we decide what to do with this.

**Comment by torkelo** [comment]
moving it to 7.1 milestone, need to look at this in the coming weeks after release, think this is super good and needed, just need to chat about where in redux it should live and if it should be part of dashboard collection you started (as panel id is only unique to single dashboard).

**Pull request closed without merging**